src/scaling/balancer.ts

```diff
@@ -0,0 +1,53 @@
+import http, { IncomingMessage, ServerResponse } from 'http';
+import cluster, { Worker } from 'cluster';
+import dotenv from 'dotenv';
+import { cpus } from 'os';
+import { pid } from 'process';
+import { configDB } from '../libs/dbHelpers';
+import { PORT } from '../utils/constants';
+import DB from '../libs/db';
+import App from '../app';
+dotenv.config();
+
+const server = http.createServer();
+const requestHandler = async (req: IncomingMessage, res: ServerResponse) => {
+  const db = new DB();
+  const app = new App(db);
+  await app.requestHandler(req, res);
+};
+
+const startServer = () => {
+  server.listen(PORT, () => {
+    console.log(`Worker is listening on the port ${PORT}`);
+  });
+  server.on('request', requestHandler);
+};
+
+try {
+  configDB.start();
+  if (cluster.isPrimary) {
+    const parallelism = cpus().length - 1;
+    console.log(`Master ${pid} is running`);
+    startServer();
```

Comment:
Unfortunately, your load balancer is not implemented. You have fork processes running starting at port 4001+. And in fact, the primary process at 4000 fulfills all requests. It was necessary to establish the server on port 4000 in the `cluster.isPrimary` block and redirect requests from it to 4001-4007 (or whatever forks we have, and forward their responses to the main port 4000.

And here another problem begins - how to store data between processes **in-memory**. Two ways:
1) Send messages between processes and store data only in cluster.isPrimary.
2) Send http requests to the main port from the child ones, using it as a database.